### PR TITLE
add Interoperability test suit

### PR DESCRIPTION
**Pull request by BusyJay**
ref: https://github.com/grpc/grpc/blob/v1.3.x/doc/interop-test-descriptions.md

### CODE REVIEW

**Comment by BusyJay** [comment]
Compression (depends on metadata), cache (needs proxy server) and Oauth/jwt/per_rpc_creds (depends on google's service) is not implemented yet.

**Comment by BusyJay** [comment]
PTAL

File: interop/src/bin/client.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        "SERVER_STREAMING" => client.server_streaming(),
+        "PING_PONG" => client.ping_pong(),
+        "EMPTY_STREAM" => client.empty_stream(),
+        "CANCEL_AFTER_BEGIN" => client.cancel_after_begin(),
+        "CANCEL_AFTER_FIRST_RESPONSE" => client.cancel_after_first_response(),
+        "TIMEOUT_ON_SLEEPING_SERVER" => client.timeout_on_sleeping_server(),
+        "STATUS_CODE_AND_MESSAGE" => client.status_code_and_message(),
+        "UNIMPLEMENTED_METHOD" => client.unimplemented_method(),
+        "UNIMPLEMENTED_SERVICE" => client.unimplemented_service(),
+        _ => panic!("unknown case: {:?}", case),
```
Comment (BusyJay) [inline interop/src/bin/client.rs]: I think they are the same.

File: interop/src/client.rs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+            .unwrap();
+        assert_eq!(resp_sizes, sizes);
+        println!("pass");
+    }
+
+    pub fn ping_pong(&self) {
+        print!("testing ping pong ... ");
+        let (mut sender, mut receiver) = self.client.full_duplex_call();
+        let cases = vec![(31415, 27182), (9, 8), (2653, 1828), (58979, 45904)];
+        for (resp_size, payload_size) in cases {
```
Comment (BusyJay) [inline interop/src/client.rs]: See L102.

Comment (overvenus) [inline interop/src/client.rs]: L102 only waits the stream, it does not check if the stream has exactly four responses.

Comment (BusyJay) [inline interop/src/client.rs]: Yes. Additional check has been added.

**Comment by BusyJay** [comment]
PTAL

**Comment by BusyJay** [comment]
@sre-bot try

**Comment by overvenus** [comment]
LGTM

**Pull request merged**